python/pyarrow/_dataset.pyx

```diff
@@ -84,7 +84,25 @@ cdef CFileSource _make_file_source(object file, FileSystem filesystem=None):
 
 
 cdef class Expression(_Weakrefable):
-
+    """
+    A logical expression to be evaluated against some input.
+
+    To create an expression:
+     - Use the factory function ``pyarrow.dataset.scalar()`` to create a
```

Comment:
Can you leave a blank line above this one (to separate the text from the list)?

And the `-` has one extra space before it as indentation, which needs to be removed in order to render it correctly with sphinx.

(restructured text is quite picky ... ;))